File: Projects/04/Factorial.java
/*
* Emily Costa
* March 4, 2020
* CS 214 A
* Project 4 - given an integer n, returns n!
*/

import java.util.Scanner;

public class Factorial {

	public static int calc_factorial (int n){

		int total = 1;

		for (int i = n; i > 0; i--){
			total *= i;
		}

		return total;
	}

	public static void main (String[] args){

		// promt user for n value
		System.out.print("\n To compute n!, enter n: ");
		// create connection to standard in
		Scanner input = new Scanner(System.in);
		// scan in input
		int n = input.nextInt();
		// call calc_factorial function and print result
		System.out.printf(" The factorial is: %d \n\n", calc_factorial(n));
	}



} // end class Factorial

File: Projects/04/factorial.clj
;;; Emily Costa
;;; March 4, 2020
;;; CS 214 A
;;; Project 4 - given an integer n, returns n!
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial) ; namespace function names the program

;;; recursive function to return n!
(defn calc_factorial [n]
	(if (= n 1)
		n
		(* n (calc_factorial(dec n)))
	) ; end if 
) ; end calc_factorial function

(defn -main []
	; prompt user and scan in inpit
	(print "\n To compute n!, enter n: ") (flush)
	(let [ n (read)]
		; call calc_factorial function and print result
		(printf " The factorial is: %d \n" (calc_factorial n))
	) ; end let

	(print "\n")
) ; end main